plasma_framework/python_tests/tests/contracts/root_chain/test_process_exits.py

```diff
@@ -717,6 +717,67 @@ def test_output_exited_via_se_and_ife_exits_only_once(testlang, plasma_framework
     assert post_exit_balance == pre_exit_balance - amount_spent
 
 
+@pytest.mark.parametrize("num_outputs", [1, 2, 3, 4])
+def test_output_exited_via_se_and_then_ife_withdraws_once(testlang, plasma_framework, num_outputs):
+    owner, amount, amount_spent = testlang.accounts[0], 100, 1
+    deposit_id = testlang.deposit(owner, amount)
+
+    outputs = []
+    for i in range(0, num_outputs):
+        outputs.append((owner.address, NULL_ADDRESS, amount_spent))
+    spend_id = testlang.spend_utxo([deposit_id], [owner], outputs)
+
+    output_index = num_outputs - 1
+
+    blknum, txindex, _ = decode_utxo_id(spend_id)
+    output_id = encode_utxo_id(blknum, txindex, output_index)
+    testlang.start_standard_exit(output_id, account=owner)
+
+    pre_exit_balance = testlang.get_balance(plasma_framework.eth_vault)
+
+    testlang.start_in_flight_exit(spend_id)
+    testlang.piggyback_in_flight_exit_output(spend_id, output_index, owner)
+    testlang.forward_timestamp(2 * MIN_EXIT_PERIOD + 1)
+    testlang.process_exits(NULL_ADDRESS, 0, 10)
+
+    post_exit_balance = testlang.get_balance(plasma_framework.eth_vault)
+    assert post_exit_balance == pre_exit_balance - amount_spent
+
+
+def test_input_exited_via_se_and_then_ife_does_not_break_exit_processing(testlang, plasma_framework):
+    owner, amount, amount_spent = testlang.accounts[0], 100, 1
+    deposit_id = testlang.deposit(owner, amount)
+
+    spend_id = testlang.spend_utxo([deposit_id], [owner], outputs=[(owner.address, NULL_ADDRESS, amount_spent)])
+
+    testlang.start_standard_exit(deposit_id, account=owner)
+
+    start_ife_piggyback_and_process(spend_id, owner, testlang)
+
+    post_exit_balance = testlang.get_balance(plasma_framework.eth_vault)
+    # deposit exit not challenged and withdrawn via se - broken Plasma
+    assert post_exit_balance == 0
```

Comment:
I would rather try to ensure that the money wasn't transferred to the exit owner.

Maybe, also, let's check events that the certain exit was omitted?

Reply:
I wouldn't go that far, funds are stealed anyway

Reply:
What I mean is that if the balance of sth vault is 0 before the processing, you will never get balance below zero (at least in eth, this may not be true to ERC20 tokens 🤔 ).

Lets make non-zero eth vault balance, before processing and then check whether sth has changed. 

And I still am for checking the event.

Reply:
right, but after line 755 it is again 0

Reply:
sure, but when processing exits we have two exits in the queue - we process the first one which transfers all the money from the contract, and then when processing the second exit the SuT may try to transfer money for the second time, but after such a transfer you will still end up with 0 amount - kind of a false positive test.

Reply:
As plasma framework does not guarantee what happens with funds after invalid exits are not challenged (se  <> ife input, both processed) I think we should not test those scenarios.